Source/Core/VideoBackends/D3D12/D3DQueuedCommandList.h

```diff
@@ -0,0 +1,618 @@
+
+	typedef struct CopyTextureRegionArguments
+	{
+		D3D12_TEXTURE_COPY_LOCATION dst;
+		// UINT DstX, DstY, DstZ; Assuming always '0'. Checked at runtime with DEBUGCHECK.
+		D3D12_TEXTURE_COPY_LOCATION src;
+		D3D12_BOX srcBox;
+	} CopyTextureRegionArguments;
+
+	typedef struct SetDescriptorHeapsArguments
+	{
+		ID3D12DescriptorHeap** ppDescriptorHeap;
+		UINT NumDescriptorHeaps;
+	} SetDescriptorHeapsArguments;
+
+	typedef struct ResourceBarrierArguments
+	{
+		D3D12_RESOURCE_BARRIER barrier;
+	} ResourceBarrierArguments;
+
+	typedef struct ResolveSubresourceArguments
+	{
+		ID3D12Resource *pDstResource;
+		UINT DstSubresource;
+		ID3D12Resource *pSrcResource;
+		UINT SrcSubresource;
+		DXGI_FORMAT Format;
+	} ResolveSubresourceArguments;
+
+	typedef struct CloseCommandListArguments
+	{
+	} CloseCommandListArguments;
+
+	typedef struct ExecuteCommandListArguments
+	{
+	} ExecuteCommandListArguments;
+
+	typedef struct PresentArguments
+	{
+		IDXGISwapChain *pSwapChain;
+		UINT SyncInterval;
+		UINT Flags;
+	} PresentArguments;
+
+	typedef struct ResetCommandListArguments
+	{
+		ID3D12CommandAllocator *pAllocator;
+	} ResetCommandListArguments;
+
+	typedef struct ResetCommandAllocatorArguments
+	{
+		ID3D12CommandAllocator *pAllocator;
+	} ResetCommandAllocatorArguments;
+
+	typedef struct FenceGpuSignalArguments
+	{
+		ID3D12Fence *pFence;
+		UINT64 fenceValue;
+	} FenceGpuSignalArguments;
+
+	typedef struct FenceCpuSignalArguments
+	{
+		ID3D12Fence *pFence;
+		UINT64 fenceValue;
+	} FenceCpuSignalArguments;
+
+	typedef struct D3DQueueItem
+	{
+		D3DQueueItemType Type;
+
+		union
+		{
+			SetPipelineStateArguments SetPipelineState;
+			SetRenderTargetsArguments SetRenderTargets;
+			SetVertexBuffersArguments SetVertexBuffers;
+			SetIndexBufferArguments SetIndexBuffer;
+			RSSetViewportsArguments RSSetViewports;
+			RSSetScissorRectsArguments RSSetScissorRects;
+			SetGraphicsRootDescriptorTableArguments SetGraphicsRootDescriptorTable;
+			SetGraphicsRootConstantBufferViewArguments SetGraphicsRootConstantBufferView;
+			SetGraphicsRootSignatureArguments SetGraphicsRootSignature;
+			ClearRenderTargetViewArguments ClearRenderTargetView;
+			ClearDepthStencilViewArguments ClearDepthStencilView;
+			DrawInstancedArguments DrawInstanced;
+			DrawIndexedInstancedArguments DrawIndexedInstanced;
+			IASetPrimitiveTopologyArguments IASetPrimitiveTopology;
+			CopyBufferRegionArguments CopyBufferRegion;
+			CopyTextureRegionArguments CopyTextureRegion;
+			SetDescriptorHeapsArguments SetDescriptorHeaps;
+			ResourceBarrierArguments ResourceBarrier;
+			ResolveSubresourceArguments ResolveSubresource;
+			CloseCommandListArguments CloseCommandList;
+			ExecuteCommandListArguments ExecuteCommandList;
+			PresentArguments Present;
+			ResetCommandListArguments ResetCommandList;
+			ResetCommandAllocatorArguments ResetCommandAllocator;
+			FenceGpuSignalArguments FenceGpuSignal;
+			FenceCpuSignalArguments FenceCpuSignal;
+		};
+	} D3DQueueItem;

```

Comment:
typedefs like this aren't necessary in C++; applies to all enums and structs above this.